### PR TITLE
join method

### PR DESCRIPTION
**Pull request by NikitaKarbazin**

### CODE REVIEW

**Review by Kaunin1 — APPROVED** [review]
Solution looks fine

File: src/arrayMethodJoin.js [hunk 1]
```diff
@@ -4,9 +4,30 @@
...
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const length = this.length;
+    let resultJoin = '';
+    let counter = 0;
+
+    while (counter < length) {
+      if (counter > 0) {
+        resultJoin += `${separator}`;
```
Comment (illiarerun) [inline src/arrayMethodJoin.js:14]: no need for extrapolation here
```suggestion
        resultJoin += separator;
```

File: src/arrayMethodJoin.js [hunk 2]
```diff
@@ -4,9 +4,30 @@
...
+      }
+
+      let next = '';
+      const elem = this[counter];
+
+      next = elem === null || elem === undefined ? '' : elem;
```
Comment (illiarerun) [inline src/arrayMethodJoin.js:20]: format ternaries properly, don't write them in one line
```suggestion
      next = elem === null || elem === undefined 
        ? '' 
        : elem;
```

Comment (illiarerun) [reply src/arrayMethodJoin.js:20]: you also don't need a let here, make it a const and set value instantly
```suggestion
      const next = ...
```

File: src/arrayMethodJoin.js [hunk 3]
```diff
@@ -4,9 +4,30 @@
...
+      counter++;
+    }
+
+    return resultJoin;
   };
 }
 
+const source = [0, 1, 2, 3];
+
+source.join(null);
```
Comment (illiarerun) [inline src/arrayMethodJoin.js:31]: remove test code
```suggestion
```

File: src/arrayMethodJoin.js [hunk 4]
```diff
@@ -4,9 +4,30 @@
...
+    while (counter < length) {
+      if (counter > 0) {
+        resultJoin += `${separator}`;
+      }
+
+      let next = '';
+      const elem = this[counter];
+
+      next = elem === null || elem === undefined ? '' : elem;
+      resultJoin += `${next}`;
```
Comment (illiarerun) [inline src/arrayMethodJoin.js:21]: no interpolation here, also
```suggestion
      resultJoin += next;
```

**Review by illiarerun — CHANGES_REQUESTED** [review]
GJ, let's work on fixes now:

**Review by illiarerun — APPROVED** [review]

**Pull request closed without merging**